include/aws/http/connection.h

```diff
@@ -35,6 +35,94 @@ typedef void(
 typedef void(
     aws_http_on_client_connection_shutdown_fn)(struct aws_http_connection *connection, int error_code, void *user_data);
 
+/*
+ * Connection target properties
+ */
+struct aws_http_endpoint_options {
+
+    /*
+     * Required
+     */
+    struct aws_byte_cursor schema;
+
+    /**
+     * Required.
+     */
+    struct aws_byte_cursor host_name;
+
+    /**
+     * Required.
+     */
+    uint16_t port;
+};
+
+/*
+ * All of the currently supported ways of connecting to an http proxy
+ */
+enum aws_http_proxy_auth_type {
+    AWS_HPAT_NAME_AND_PASS
+};
+
+/*
+ * Configuration for username/password based proxy authentication
+ */
+struct aws_http_client_proxy_name_and_pass_options {
+
+    /*
+     * Required
+     */
+    struct aws_byte_cursor username;
+
+    /*
+     * Required
+     */
+    struct aws_byte_cursor password;
+};
+
+/*
+ * Optional configuration for proxy connection authorization.  At the moment, only supports
+ * standard username/password auth via Proxy-Authorization header.
+ */
+struct aws_http_client_proxy_auth_options {
+
+    enum aws_http_proxy_auth_type type;
+
+    union {
+        struct aws_http_client_proxy_name_and_pass_options name_and_pass_options;
```

Comment:
Are there other headers someone might want to add to the CONNECT request?
Would this ever be signed?
Those might be arguments for just accepting a list of headers and not doing anything fancy beyond that